x-pack/plugins/infra/public/components/waffle/index.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import styled from 'styled-components';
+import { AutoSizer } from '../auto_sizer';
+import { GroupOfGroups } from './group_of_groups';
+import { GroupOfNodes } from './group_of_nodes';
+
+import {
+  isWaffleMapGroupWithGroups,
+  isWaffleMapGroupWithNodes,
+} from '../../containers/libs/type_guards';
+import { InfraWaffleData, InfraWaffleMapGroup, InfraWaffleOptions } from '../../lib/lib';
+import { applyWaffleMapLayout } from './lib/apply_wafflemap_layout';
+import { Scroll } from './scroll';
+
+interface WafleMapProps {
+  options: InfraWaffleOptions;
+  map: InfraWaffleData;
+}
+
+const initialState = {
+  hasScroll: false,
+};
+
+type WaffleMapState = Readonly<typeof initialState>;
+
+export class Waffle extends React.Component<WafleMapProps, WaffleMapState> {
+  public readonly state: WaffleMapState = initialState;
+  private outer: any;
+  private inner: any;
+  private scrollBarTimer: any;
+
+  public render() {
+    return (
+      <AutoSizer bounds>
+        {({ measureRef, bounds: { width = 0, height = 0 } }) => {
```

Comment:
Maybe we want to use `content` (measurements of the content box) instead of `bounds` (measurement of the border box) to more accurately reflect the space available to the waffle map?

![image](https://user-images.githubusercontent.com/973741/43836012-3eb17d4e-9b14-11e8-96ea-b0f0b2cbb7c1.png)
(source https://developers.google.com/web/updates/2016/10/resizeobserver)